src/Interpreters/Streaming/WindowCommon.h

```diff
@@ -273,6 +273,9 @@ struct SessionWindowParams : WindowParams
     bool start_with_inclusion;
     bool end_with_inclusion;
 
+    /// TODO: So far, always assign session window in aggr
+    bool assign_window_pushdown = true;
```

Comment:
```
/// So far, only for session window, we evaluate the watermark for the events in Aggregate Transform
/// For other windows, we assigned the watermark in Watermark stamper step.
```

Reply:
Previously we are kind mixing `window assignment and watermark assignment` for tumble / hopping / global windows since this is probably more performant.

Right now, we are separating these 2 steps, which are good for maintainability moving forward.

Reply:
For session window, fundamentally, it doesn't have a fixed watermark because it highly depends on inputs and the evaluation of start / end expression.

So, it seems window assignment for now all happens at Aggregate transform step ? The Watermark evaluation happens in WatermarkStamper step and for session window, we don't have watermark

Reply:
`pushdown_window_assignment`